### PR TITLE
SY-1647 labjack error on all disabled channels in task

### PR DESCRIPTION
**Pull request by Lham42**
# Issue Pull Request

## Key Information

- **Linear Issue**: [SY-1647](https://linear.app/synnax/issue/1647)

## Description

This PR checks if channels are enabled and returns an error on configure if you enabled channels are passed in. Also refactors some of the error logging in labjack tasks. 

## Basic Readiness

- [ ] I have performed a self-review of my code.
- [ ] I have added relevant tests to cover the changes to CI.
- [ ] I have added needed QA steps to the [release candidate](/synnaxlabs/synnax/blob/main/.github/PULL_REQUEST_TEMPLATE/rc.md) template that cover these changes.
- [ ] I have updated in-code documentation to reflect the changes.
- [ ] I have updated user-facing documentation to reflect the changes.

### Data Structures

I have ensured that previous versions of stored data structures are properly migrated to new formats in the following projects:

- [ ] Server
- [ ] Console

### API Changes

The following projects have backwards-compatible APIs:

- [ ] Python Client
- [ ] Server
- [ ] TypeScript Client

### CODE REVIEW

File: driver/labjack/writer_sink.cpp [hunk 1]
```diff
@@ -251,15 +256,14 @@ std::vector<synnax::ChannelKey> labjack::WriteSink::get_state_channel_keys() {
...
-        LOG(ERROR) << "[labjack.writer] No channels configured";
         return {};
     }
 
     std::set<synnax::ChannelKey> unique_keys;
     for (auto &channel: this->writer_config.channels) {
         auto [channel_info, err] = this->ctx->client->channels.retrieve(channel.state_key);
         if (err) {
-            LOG(ERROR) << "[labjack.writer] Failed to retrieve channel: " << channel.state_key;
+            this->log_err("Failed to retrieve channel.");
```
Comment (emilbon99) [inline driver/labjack/writer_sink.cpp:266]: This message gets returned to the caller, so it needs to be more detailed. What port is the problem in? User needs as much info as possible.

File: driver/labjack/writer.h [hunk 2]
```diff
@@ -135,7 +135,8 @@ struct WriterConfig {
                 return;
             }
             channel.data_type = channel_info.data_type;
-            channels.emplace_back(channel);
+
+            if(channel.enabled) channels.emplace_back(channel);
```
Comment (emilbon99) [inline driver/labjack/writer.h:139]: Fix formatting. Also, why are you still retrieving the channel and updating it in the initial state map if it is disabled? Why not just check if the channel is disabled on 130 and return early?

File: driver/labjack/reader_task.cpp [hunk 3]
```diff
@@ -114,6 +114,9 @@ std::unique_ptr<task::Task> labjack::ReaderTask::configure(
         breaker_config
     );
 
+    if (!source->ok())
```
Comment (emilbon99) [inline driver/labjack/reader_task.cpp:117]: Inline this.

File: driver/labjack/reader_source.cpp [hunk 4]
```diff
@@ -506,3 +524,16 @@ std::vector<synnax::ChannelKey> labjack::ReaderSource::get_ai_channel_keys() {
     }
     return keys;
 }
+
+void labjack::ReaderSource::log_err(std::string msg){
+    LOG(ERROR) << "[labjack.reader] " << msg;
+    this->ok_state = false;
+    ctx->set_state({
+                       .task = this->task.key,
```
Comment (emilbon99) [inline driver/labjack/reader_source.cpp:532]: This formatting looks bad

File: driver/labjack/reader_source.cpp [hunk 5]
```diff
@@ -85,30 +90,28 @@ void labjack::ReaderSource::init() {
...
     this->init_tcs();
 }
 
 
 void labjack::ReaderSource::init_tcs() {
-    if (this->reader_config.tc_channels.empty()) {
-        return;
-    }
+    if (this->reader_config.tc_channels.empty()) return;
+    if(this->reader_config.device_type == "T4")
```
Comment (emilbon99) [inline driver/labjack/reader_source.cpp:113]: Why don't we support TCs on a T4? Was this always the case?

File: driver/labjack/writer_sink.cpp (hunk 1 above)
Comment (emilbon99) [reply driver/labjack/writer_sink.cpp:266]: Also address this for the other channel retrieval calls.

**Review by emilbon99 — CHANGES_REQUESTED** [review]

File: driver/labjack/writer_sink.cpp (hunk 1 above)
Comment (Lham42) [reply driver/labjack/writer_sink.cpp:266]: resolved

File: driver/labjack/reader_source.cpp (hunk 5 above)
Comment (Lham42) [reply driver/labjack/reader_source.cpp:113]: Yes. T4 does not have the hardware built in to support thermocouple channels

File: driver/labjack/writer.h (hunk 2 above)
Comment (Lham42) [reply driver/labjack/writer.h:139]: resolved

**Review by emilbon99 — APPROVED** [review]
Approved after formatting issue gets fixed

**Pull request merged**